tests/v2/validation/steve/vai/secret_filter_test_cases.go

```diff
@@ -0,0 +1,286 @@
+			return url.Values{
+				"filter":               []string{"metadata.name=secret1"},
+				"projectsornamespaces": namespaces,
+			}
+		},
+		supportedWithVai: true,
+	},
+	{
+		name: "Filter by single label",
+		createSecrets: func() ([]v1.Secret, []string, []string, []string) {
+			suffix := namegen.RandStringLower(randomStringLength)
+			ns := fmt.Sprintf("namespace-%s", suffix)
+			name1 := fmt.Sprintf("secret1-%s", namegen.RandStringLower(randomStringLength))
+			name2 := fmt.Sprintf("secret2-%s", namegen.RandStringLower(randomStringLength))
+
+			secrets := []v1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name1,
+						Namespace: ns,
+						Labels:    map[string]string{"key1": "value1"},
+					},
+					Type: v1.SecretTypeOpaque,
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name2,
+						Namespace: ns,
+						Labels:    map[string]string{"key1": "value2"},
+					},
+					Type: v1.SecretTypeOpaque,
+				},
+			}
+
+			expectedNames := []string{name1}
+			allNamespaces := []string{ns}
+			expectedNamespaces := []string{ns}
+
+			return secrets, expectedNames, allNamespaces, expectedNamespaces
+		},
+		filter: func(namespaces []string) url.Values {
+			return url.Values{
+				"filter":               []string{"metadata.labels.key1=value1"},
+				"projectsornamespaces": namespaces,
+			}
+		},
+		supportedWithVai: false,
+	},
+	{
+		name: "Filter by multiple labels",
+		createSecrets: func() ([]v1.Secret, []string, []string, []string) {
+			suffix := namegen.RandStringLower(randomStringLength)
+			ns := fmt.Sprintf("namespace-%s", suffix)
+			name1 := fmt.Sprintf("secret1-%s", namegen.RandStringLower(randomStringLength))
+			name2 := fmt.Sprintf("secret2-%s", namegen.RandStringLower(randomStringLength))
+			name3 := fmt.Sprintf("secret3-%s", namegen.RandStringLower(randomStringLength))
+
+			secrets := []v1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name1,
+						Namespace: ns,
+						Labels:    map[string]string{"key1": "value1", "key2": "value2"},
+					},
+					Type: v1.SecretTypeOpaque,
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name2,
+						Namespace: ns,
+						Labels:    map[string]string{"key1": "value1"},
+					},
+					Type: v1.SecretTypeOpaque,
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      name3,
+						Namespace: ns,
+						Labels:    map[string]string{"key2": "value2"},
+					},
+					Type: v1.SecretTypeOpaque,
+				},
+			}
+
+			expectedNames := []string{name1}
+			allNamespaces := []string{ns}
+			expectedNamespaces := []string{ns}
+
+			return secrets, expectedNames, allNamespaces, expectedNamespaces
+		},
+		filter: func(namespaces []string) url.Values {
+			return url.Values{
+				"filter":               []string{"metadata.labels.key1=value1", "metadata.labels.key2=value2"},
+				"projectsornamespaces": namespaces,
+			}
+		},
+		supportedWithVai: false,
```

Comment:
curious, why is this false?

Reply:
This isn't supported with vai currently, it's detailed in the design doc. This functionality will eventually be added, then this bool can be changed when that PR comes in at a later release